### PR TITLE
Migrate to Yarn 2+ and bump dependencies

### PR DESCRIPTION
**Pull request by Discookie**
Currently the build seems to fail on CI because of an outdated webpack using an outdated OpenSSL version.

Updated ESLint dependencies, and using Yarn 2 lockfiles now.

### CODE REVIEW

File: src/test/runTest.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { spawnSync } from 'child_process';
 import * as path from 'path';
 import * as fs from 'fs';
 
-import { runTests } from 'vscode-test';
+import { runTests } from '@vscode/test-electron';
```
Comment (vodorok) [inline src/test/runTest.ts:5]: Just a question, is this related to the Yarn2 upgrade?

**Review by vodorok — APPROVED** [review]
LGTM apart from one question.

File: src/test/runTest.ts (hunk 1 above)
Comment (Discookie) [reply src/test/runTest.ts:5]: The package was apparently migrated it to a different namespace, and the old repository is now deprecated. Same happened with `vsce` => `@vscode/vsce`. Yarn gave warnings about the old ones.

Comment (Discookie) [reply src/test/runTest.ts:5]: (This was a general issue, not a Yarn 2-specific one)
